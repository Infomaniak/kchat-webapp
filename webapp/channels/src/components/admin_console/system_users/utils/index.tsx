// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {SortingState} from '@tanstack/react-table';
import React from 'react';
import {FormattedMessage} from 'react-intl';

import {UserReportSortColumns, ReportSortDirection} from '@mattermost/types/reports';
import type {UserReportOptions, UserReport} from '@mattermost/types/reports';
import type {Team} from '@mattermost/types/teams';

import {PAGE_SIZES} from 'components/admin_console/list_table';

import type {AdminConsoleUserManagementTableProperties} from 'types/store/views';

import {ColumnNames, RoleFilters, StatusFilter, TeamFilters} from '../constants';
import type {TableOptions} from '../system_users';
import type {OptionType as TeamFilterOptionType} from '../system_users_filters_popover/system_users_filter_team';

export function convertTableOptionsToUserReportOptions(tableOptions?: TableOptions): UserReportOptions {
    return {
        page_size: tableOptions?.pageSize || PAGE_SIZES[0],
        from_column_value: tableOptions?.fromColumnValue,
        from_id: tableOptions?.fromId,
        direction: tableOptions?.direction,
        ...getSortColumnForOptions(tableOptions?.sortColumn),
        ...getSortDirectionForOptions(tableOptions?.sortIsDescending),
        ...getSearchFilterOption(tableOptions?.searchTerm),
        ...getTeamFilterOption(tableOptions?.filterTeam),
        ...getStatusFilterOption(tableOptions?.filterStatus),
        ...getRoleFilterOption(tableOptions?.filterRole),
        date_range: tableOptions?.dateRange,
    };
}

/**
 * Converts the sorting column name to API compatible sorting column name. Default sorting column name is by username.
 */
export function getSortColumnForOptions(id?: SortingState[0]['id']): Pick<UserReportOptions, 'sort_column'> {
    let sortColumn: UserReportOptions['sort_column'];

    if (id === ColumnNames.email) {
        sortColumn = UserReportSortColumns.email;
    } else if (id === ColumnNames.createAt) {
        sortColumn = UserReportSortColumns.createAt;
    } else {
        // Default sorting to first User details column
        sortColumn = UserReportSortColumns.username;
    }

    return {
        sort_column: sortColumn,
    };
}

/**
 * Converts the sorting direction to API compatible sorting direction. Default sorting direction is ascending.
 */
export function getSortDirectionForOptions(desc?: SortingState[0]['desc']): Pick<UserReportOptions, 'sort_direction'> {
    let sortDirection: UserReportOptions['sort_direction'];

    if (desc) {
        sortDirection = ReportSortDirection.descending;
    } else {
        sortDirection = ReportSortDirection.ascending;
    }

    return {
        sort_direction: sortDirection,
    };
}

/**
 * It returns the value of that column on which sorting is applied.
 */
export function getSortableColumnValueBySortColumn(row: UserReport, sortColumn: AdminConsoleUserManagementTableProperties['sortColumn']): string {
    switch (sortColumn) {
    case ColumnNames.email:
        return row.email;
    case ColumnNames.createAt:
        return String(row.create_at);
    default:
        return row.username;
    }
}

export function getTeamFilterOption(teamId?: string): Partial<Pick<UserReportOptions, 'team_filter' | 'has_no_team'>> {
    if (!teamId || teamId === TeamFilters.AllTeams) {
        return {
            team_filter: undefined,
            has_no_team: undefined,
        };
    } else if (teamId === TeamFilters.NoTeams) {
        return {
            team_filter: undefined,
            has_no_team: true,
        };
    }

    return {
        team_filter: teamId,
        has_no_team: undefined,
    };
}

export function getStatusFilterOption(status?: string): Partial<Pick<UserReportOptions, 'hide_active' | 'hide_inactive'>> {
    if (status === StatusFilter.Active) {
        return {
            hide_inactive: true,
        };
    } else if (status === StatusFilter.Deactivated) {
        return {
            hide_active: true,
        };
    }

    return {
        hide_active: undefined,
        hide_inactive: undefined,
    };
}

export function getPaginationInfo(pageIndex: number, pageSize: number, currentLength: number, total?: number) {
    if (!currentLength) {
        return (
            <FormattedMessage
                id='admin.system_users_list.pagination.no_users'
                defaultMessage='0 users'
            />
        );
    }

    const firstPage = (pageIndex * pageSize) + 1;
    const lastPage = (pageIndex * pageSize) + currentLength;
    const totalItems = total || 0;

    return (
        <FormattedMessage
            id='admin.system_users_list.pagination'
            defaultMessage='Showing {firstPage} - {lastPage} of {totalItems} users'
            values={{
                firstPage,
                lastPage,
                totalItems,
            }}
        />
    );
}

export function getDefaultSelectedValueFromList<T extends {value: string}>(value: string, options: T[]) {
    const option = options.find((option) => option.value === value);

    if (option) {
        return option;
    }

    return options[0];
}

export function getDefaultSelectedTeam(teamId: Team['id'] | string, label?: string): TeamFilterOptionType {
    if (!teamId || teamId === TeamFilters.AllTeams) {
        return {
            value: TeamFilters.AllTeams,
            label: (
                <FormattedMessage
                    id='admin.system_users.filters.team.allTeams'
                    defaultMessage='All Teams'
                />
            ),
        };
    } else if (teamId === TeamFilters.NoTeams) {
        return {
            value: TeamFilters.NoTeams,
            label: (
                <FormattedMessage
                    id='admin.system_users.filters.team.noTeams'
                    defaultMessage='No Teams'
                />
            ),
        };
    }

    return {
        value: teamId,
        label: label || '',
    };
}

export function getRoleFilterOption(role?: string): Pick<UserReportOptions, 'role_filter'> {
    if (!role || role === RoleFilters.Any) {
        return {role_filter: undefined};
    }
    return {role_filter: role};
}

export function getSearchFilterOption(search?: string): Pick<UserReportOptions, 'search_term'> {
    if (!search || search.trim().length === 0) {
        return {search_term: undefined};
    }

    return {search_term: search};
}
