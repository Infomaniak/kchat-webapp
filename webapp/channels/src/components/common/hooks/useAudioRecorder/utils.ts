export const getWasmFileURL = () =>
    new URL('wasm-media-encoders/wasm/mp3', import.meta.url);
