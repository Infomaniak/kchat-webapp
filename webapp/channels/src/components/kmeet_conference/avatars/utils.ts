export function countMeta<T>(
    items: T[],
    total = items.length,
): [T[], T[], {overflowUnnamedCount: number; nonDisplayCount: number}] {
    const breakAt = Math.max(items.length, total) > 4 ? 3 : 4;

    const displayItems = items.slice(0, breakAt);
    const overflowItems = items.slice(breakAt);

    const overflowUnnamedCount = Math.max(total - displayItems.length - overflowItems.length, 0);
    const nonDisplayCount = overflowItems.length + overflowUnnamedCount;

    return [displayItems, overflowItems, {overflowUnnamedCount, nonDisplayCount}];
}
