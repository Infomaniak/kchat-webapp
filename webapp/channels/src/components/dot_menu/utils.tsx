// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type React from 'react';
import {EventTypes, TELEMETRY_CATEGORIES} from 'utils/constants';

import {trackEvent} from 'actions/telemetry_actions';

export type ChangeEvent = React.KeyboardEvent | React.MouseEvent;

export const trackDotMenuEvent = (e: ChangeEvent, suffix: string): void => {
    if (e.type === EventTypes.CLICK) {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.CLICK + '_' + suffix);
    } else {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.SHORTCUT + '_ ' + suffix);
    }
};
