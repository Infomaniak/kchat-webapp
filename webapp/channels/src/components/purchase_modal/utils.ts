// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {Product} from '@mattermost/types/cloud';

import {CloudProducts, RecurringIntervals} from 'utils/constants';

/**
 *
 * @param products  Record<string, Product> | undefined - the list of current cloud products
 * @param productId String - a valid product id used to find a particular product in the dictionary
 * @param productSku String - the sku value of the product of type either cloud-starter | cloud-professional | cloud-enterprise
 * @returns Product
 */
export function findProductInDictionary(products: Record<string, Product> | undefined, productId?: string | null, productSku?: string, productRecurringInterval?: string): Product | null {
    if (!products) {
        return null;
    }
    const keys = Object.keys(products);
    if (!keys.length) {
        return null;
    }
    if (!productId && !productSku) {
        return products[keys[0]];
    }
    let currentProduct = products[keys[0]];
    if (keys.length > 1) {
        // here find the product by the provided id or name, otherwise return the one with Professional in the name
        keys.forEach((key) => {
            if (productId && products[key].id === productId) {
                currentProduct = products[key];
            } else if (productSku && products[key].sku === productSku && products[key].recurring_interval === productRecurringInterval) {
                currentProduct = products[key];
            }
        });
    }
    return currentProduct;
}

export function getSelectedProduct(yearlyProducts: Record<string, Product>) {
    return findProductInDictionary(yearlyProducts, null, CloudProducts.PROFESSIONAL, RecurringIntervals.YEAR);
}
