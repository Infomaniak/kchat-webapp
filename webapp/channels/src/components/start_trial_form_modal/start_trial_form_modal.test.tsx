// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {BrowserRouter} from 'react-router-dom';

import type {DeepPartial} from '@mattermost/types/utilities';

import {trackEvent} from 'actions/telemetry_actions';

import {
    renderWithContext,
    screen,
    waitFor,
} from 'tests/react_testing_utils';
import {ModalIdentifiers} from 'utils/constants';

import type {GlobalState} from 'types/store';

import StartTrialFormModal from '.';

jest.mock('actions/telemetry_actions.jsx', () => {
    const original = jest.requireActual('actions/telemetry_actions.jsx');
    return {
        ...original,
        trackEvent: jest.fn(),
    };
});

describe('components/start_trial_form_modal/start_trial_form_modal', () => {
    const state: DeepPartial<GlobalState> = {
        entities: {
            users: {
                currentUserId: 'user1',
                profiles: {
                    user1: {
                        id: 'user1',
                        roles: '',
                        email: 'test@mattermost.com',
                    },
                },
            },
            general: {
                license: {
                    IsLicensed: 'true',
                    Cloud: 'false',
                },
                config: {
                    TelemetryId: 'test123',
                },
            },
        },
        views: {
            modals: {
                modalState: {
                    [ModalIdentifiers.START_TRIAL_FORM_MODAL]: {
                        open: true,
                    },
                },
            },
            admin: {
                navigationBlock: {
                    blocked: true,
                },
            },
        },
    };

    const handleOnClose = jest.fn();

    const props = {
        onClose: handleOnClose,
        page: 'some_modal',
    };

    test('should match snapshot', async () => {
        const wrapper = await waitFor(() => {
            return renderWithContext(
                <BrowserRouter>
                    <StartTrialFormModal {...props}/>
                </BrowserRouter>,
                state,
            );
        });
        expect(wrapper!).toMatchSnapshot();
    });

    test('should pre-fill email, fire trackEvent', () => {
        waitFor(() => {
            renderWithContext(
                <BrowserRouter>
                    <StartTrialFormModal {...props}/>
                </BrowserRouter>,
                state,
            );
        });
        expect(screen.getByDisplayValue('test@mattermost.com')).toBeInTheDocument();
        expect(trackEvent).toHaveBeenCalled();
    });

    test('Start trial button should be disabled on load', () => {
        waitFor(() => {
            renderWithContext(
                <BrowserRouter>
                    <StartTrialFormModal {...props}/>
                </BrowserRouter>,
                state,
            );
        });
        expect(screen.getByRole('button', {name: 'Start trial'})).toBeDisabled();
    });
});
