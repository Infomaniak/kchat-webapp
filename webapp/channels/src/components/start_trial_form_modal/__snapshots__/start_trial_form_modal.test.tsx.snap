// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/start_trial_form_modal/start_trial_form_modal should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    style="padding-right: 0px;"
  >
    <div
      aria-hidden="true"
    />
    <div
      role="dialog"
    >
      <div
        class="fade modal-backdrop in"
      />
      <div
        class="fade AirGappedModal in modal"
        id="airGappedModal"
        role="dialog"
        style="display: block; padding-right: 0px;"
        tabindex="-1"
      >
        <div
          class="AirGappedModal__dialog modal-dialog"
        >
          <div
            class="modal-content"
            role="document"
          >
            <div
              class="modal-header"
            >
              <button
                aria-label="Close"
                class="close"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  Ã—
                </span>
                <span
                  class="sr-only"
                >
                  Close
                </span>
              </button>
              <div
                class="title"
              >
                Request a trial key
              </div>
            </div>
            <div
              class="modal-body"
            >
              <div
                class="body"
              >
                <div
                  class="description"
                >
                  To start your trial, please visit 
                  <a
                    href="https://mattermost.com/trial/"
                    location="start_trial_air_gapped_modal"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    https://mattermost.com/trial/
                  </a>
                   and request a trial key.
                </div>
                <div
                  class="icon"
                >
                  <svg
                    fill="none"
                    height="200"
                    viewBox="0 0 223 159"
                    width="256"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      clip-path="url(#clip0_754_57919)"
                    >
                      <path
                        d="M17.1671 115.779H174.208C175.364 115.763 176.466 115.281 177.273 114.439C178.081 113.597 178.528 112.464 178.516 111.287V4.46575C178.528 3.28926 178.081 2.15606 177.273 1.3142C176.466 0.47233 175.364 -0.00961028 174.208 -0.026123H17.1671C16.0121 -0.00797133 14.9109 0.474468 14.1039 1.31597C13.297 2.15748 12.8496 3.28974 12.8595 4.46575V111.294C12.8513 112.468 13.2993 113.599 14.1061 114.439C14.9129 115.279 16.0131 115.761 17.1671 115.779Z"
                        fill="#3F4350"
                      />
                      <path
                        d="M-0.15625 131.141C-0.15625 136.459 4.17094 141.776 9.41922 141.776H181.963C186.909 141.776 191.532 136.471 191.532 131.141H-0.15625Z"
                        fill="#767D93"
                      />
                      <path
                        d="M175.559 115.779H15.8173L-0.15625 131.141H191.532L175.559 115.779Z"
                        fill="#D1D4DB"
                      />
                      <path
                        d="M170.782 116.961H20.5884L15.2266 123.46H176.15L170.782 116.961Z"
                        fill="#AFB3C0"
                      />
                      <path
                        d="M112.155 125.823H79.2272L76.7559 129.369H114.62L112.155 125.823Z"
                        fill="#24262E"
                      />
                      <g
                        clip-path="url(#clip1_754_57919)"
                      >
                        <rect
                          fill="white"
                          height="94.5349"
                          transform="translate(22.3262 10.0183)"
                          width="146.725"
                        />
                        <rect
                          fill="#1E325C"
                          height="95.1258"
                          width="43.7808"
                          x="22.3262"
                          y="10.0183"
                        />
                        <ellipse
                          cx="29.4256"
                          cy="17.6994"
                          fill="white"
                          opacity="0.32"
                          rx="4.14142"
                          ry="4.1359"
                        />
                        <rect
                          fill="white"
                          height="1.77253"
                          opacity="0.32"
                          rx="0.886265"
                          width="19.5238"
                          x="36.5254"
                          y="16.5176"
                        />
                        <rect
                          fill="white"
                          height="1.77253"
                          opacity="0.32"
                          rx="0.886265"
                          width="13.0159"
                          x="26.4678"
                          y="27.1528"
                        />
                        <rect
                          fill="white"
                          height="1.77253"
                          opacity="0.32"
                          rx="0.886265"
                          width="13.0159"
                          x="26.4678"
                          y="83.8738"
                        />
                        <ellipse
                          cx="28.5385"
                          cy="34.5382"
                          fill="white"
                          opacity="0.32"
                          rx="2.07071"
                          ry="2.06795"
                        />
                        <rect
                          fill="white"
                          height="1.69648"
                          opacity="0.32"
                          rx="0.848242"
                          width="24.5527"
                          x="32.9756"
                          y="33.6519"
                        />
                        <ellipse
                          cx="28.5385"
                          cy="55.8087"
                          fill="white"
                          opacity="0.32"
                          rx="2.07071"
                          ry="2.06795"
                        />
                        <rect
                          fill="white"
                          height="1.77253"
                          opacity="0.32"
                          rx="0.886265"
                          width="24.2569"
                          x="32.9756"
                          y="54.9224"
                        />
                        <ellipse
                          cx="28.5385"
                          cy="41.6285"
                          fill="white"
                          opacity="0.32"
                          rx="2.07071"
                          ry="2.06795"
                        />
                        <rect
                          fill="white"
                          height="1.77253"
                          opacity="0.32"
                          rx="0.886264"
                          width="21.2987"
                          x="32.9756"
                          y="40.7422"
                        />
                        <ellipse
                          cx="28.5385"
                          cy="62.8988"
                          fill="white"
                          opacity="0.32"
                          rx="2.07071"
                          ry="2.06795"
                        />
                        <rect
                          fill="white"
                          height="1.77253"
                          opacity="0.32"
                          rx="0.886265"
                          width="21.2987"
                          x="32.9756"
                          y="62.0125"
                        />
                        <ellipse
                          cx="28.5385"
                          cy="48.7186"
                          fill="white"
                          opacity="0.32"
                          rx="2.07071"
                          ry="2.06795"
                        />
                        <rect
                          fill="white"
                          height="1.77253"
                          opacity="0.32"
                          rx="0.886265"
                          width="27.8067"
                          x="32.9756"
                          y="47.8323"
                        />
                        <ellipse
                          cx="28.5385"
                          cy="69.9888"
                          fill="white"
                          opacity="0.32"
                          rx="2.07071"
                          ry="2.06795"
                        />
                        <rect
                          fill="white"
                          height="1.77253"
                          opacity="0.32"
                          rx="0.886265"
                          width="27.8067"
                          x="32.9756"
                          y="69.1025"
                        />
                        <ellipse
                          cx="28.5385"
                          cy="91.2594"
                          fill="white"
                          opacity="0.32"
                          rx="2.07071"
                          ry="2.06795"
                        />
                        <rect
                          fill="white"
                          height="1.77253"
                          opacity="0.32"
                          rx="0.886265"
                          width="27.8067"
                          x="32.9756"
                          y="90.373"
                        />
                        <ellipse
                          cx="28.5385"
                          cy="77.0789"
                          fill="white"
                          opacity="0.32"
                          rx="2.07071"
                          ry="2.06795"
                        />
                        <rect
                          fill="white"
                          height="1.77253"
                          opacity="0.32"
                          rx="0.886265"
                          width="18.3406"
                          x="32.9756"
                          y="76.1926"
                        />
                        <ellipse
                          cx="74.3896"
                          cy="34.8336"
                          fill="#3F4350"
                          fill-opacity="0.24"
                          rx="3.54979"
                          ry="3.54506"
                        />
                        <rect
                          fill="#3F4350"
                          fill-opacity="0.24"
                          height="1.77253"
                          rx="0.886265"
                          width="20.1155"
                          x="81.4893"
                          y="31.8794"
                        />
                        <rect
                          fill="#3F4350"
                          fill-opacity="0.24"
                          height="1.18169"
                          rx="0.590843"
                          width="60.9381"
                          x="81.4893"
                          y="37.197"
                        />
                        <rect
                          fill="#3F4350"
                          fill-opacity="0.24"
                          height="1.18169"
                          rx="0.590843"
                          width="68.6293"
                          x="81.4893"
                          y="41.9238"
                        />
                        <ellipse
                          cx="74.3896"
                          cy="59.0582"
                          fill="#3F4350"
                          fill-opacity="0.24"
                          rx="3.54979"
                          ry="3.54506"
                        />
                        <rect
                          fill="#3F4350"
                          fill-opacity="0.24"
                          height="1.77253"
                          rx="0.886265"
                          width="20.1155"
                          x="81.4893"
                          y="56.104"
                        />
                        <rect
                          fill="#3F4350"
                          fill-opacity="0.24"
                          height="1.18169"
                          rx="0.590843"
                          width="60.9381"
                          x="81.4893"
                          y="61.4216"
                        />
                        <rect
                          fill="#3F4350"
                          fill-opacity="0.24"
                          height="1.18169"
                          rx="0.590843"
                          width="68.6293"
                          x="81.4893"
                          y="66.1484"
                        />
                        <ellipse
                          cx="74.3896"
                          cy="81.5104"
                          fill="#3F4350"
                          fill-opacity="0.24"
                          rx="3.54979"
                          ry="3.54506"
                        />
                        <rect
                          fill="#3F4350"
                          fill-opacity="0.24"
                          height="1.77253"
                          rx="0.886265"
                          width="20.1155"
                          x="81.4893"
                          y="78.5562"
                        />
                        <rect
                          fill="#3F4350"
                          fill-opacity="0.24"
                          height="1.18169"
                          rx="0.590843"
                          width="60.9381"
                          x="81.4893"
                          y="83.8738"
                        />
                        <rect
                          fill="#3F4350"
                          fill-opacity="0.24"
                          height="1.18169"
                          rx="0.590843"
                          width="68.6293"
                          x="81.4893"
                          y="88.6006"
                        />
                        <ellipse
                          cx="92.73"
                          cy="17.6993"
                          fill="#3F4350"
                          fill-opacity="0.24"
                          rx="1.77489"
                          ry="1.77253"
                        />
                        <ellipse
                          cx="124.086"
                          cy="17.6993"
                          fill="#3F4350"
                          fill-opacity="0.24"
                          rx="1.77489"
                          ry="1.77253"
                        />
                        <ellipse
                          cx="154.26"
                          cy="17.6993"
                          fill="#3F4350"
                          fill-opacity="0.24"
                          rx="1.77489"
                          ry="1.77253"
                        />
                        <ellipse
                          cx="159.584"
                          cy="17.6993"
                          fill="#3F4350"
                          fill-opacity="0.24"
                          rx="1.77489"
                          ry="1.77253"
                        />
                        <ellipse
                          cx="97.4634"
                          cy="17.6993"
                          fill="#3F4350"
                          fill-opacity="0.24"
                          rx="1.77489"
                          ry="1.77253"
                        />
                        <rect
                          fill="#3F4350"
                          fill-opacity="0.24"
                          height="1.18169"
                          rx="0.590843"
                          width="18.3406"
                          x="70.8398"
                          y="17.1084"
                        />
                        <rect
                          fill="#3F4350"
                          fill-opacity="0.24"
                          height="2.36337"
                          rx="1.18169"
                          width="23.0736"
                          x="127.637"
                          y="16.5176"
                        />
                      </g>
                      <path
                        d="M95.6878 2.92798C96.1559 2.92798 96.6134 3.06659 97.0026 3.32629C97.3918 3.58598 97.6951 3.95508 97.8742 4.38693C98.0533 4.81878 98.1002 5.29397 98.0089 5.75242C97.9175 6.21087 97.6922 6.63199 97.3612 6.96251C97.0303 7.29303 96.6086 7.51813 96.1495 7.60932C95.6905 7.70051 95.2146 7.65369 94.7822 7.47481C94.3497 7.29593 93.9801 6.99303 93.7201 6.60438C93.4601 6.21572 93.3213 5.75878 93.3213 5.29135C93.3213 4.66455 93.5707 4.06341 94.0145 3.62019C94.4583 3.17697 95.0602 2.92798 95.6878 2.92798Z"
                        fill="#989DAE"
                      />
                      <path
                        d="M109.322 138.231H81.4444C80.4823 138.231 77.9395 138.231 77.9395 134.686H112.846C112.846 138.231 110.228 138.231 109.322 138.231Z"
                        fill="#3F4350"
                      />
                    </g>
                    <g
                      clip-path="url(#clip2_754_57919)"
                    >
                      <rect
                        fill="#8D93A5"
                        height="2.45026"
                        rx="1.22513"
                        width="14.3123"
                        x="184.015"
                        y="147.833"
                      />
                      <path
                        clip-rule="evenodd"
                        d="M219.65 155.56C218.049 157.196 215.767 158.45 213.264 158.45H169.486C166.983 158.45 164.701 157.196 163.1 155.56C161.496 153.92 160.297 151.613 160.297 149.117V47.721C160.297 45.2239 161.494 42.9173 163.098 41.2769C164.698 39.6401 166.979 38.3875 169.482 38.3875H213.264C215.766 38.3875 218.048 39.6398 219.65 41.2762C221.254 42.9162 222.453 45.223 222.453 47.721V149.117C222.453 151.613 221.254 153.92 219.65 155.56ZM213.264 156C216.632 156 219.999 152.554 219.999 149.117V47.721C219.999 44.2793 216.632 40.8377 213.264 40.8377H169.482C166.114 40.8377 162.75 44.2793 162.75 47.721V149.117C162.75 152.554 166.118 156 169.486 156H213.264Z"
                        fill="#363A45"
                        fill-rule="evenodd"
                      />
                      <path
                        d="M220 149.117C220 152.554 216.632 156 213.265 156H169.486C166.119 156 162.751 152.554 162.751 149.117V47.7209C162.751 44.2793 166.115 40.8376 169.482 40.8376H213.265C216.632 40.8376 220 44.2793 220 47.7209V149.117Z"
                        fill="#3F4350"
                      />
                      <path
                        d="M196.283 46.7579C196.283 47.0974 196.068 47.3719 195.796 47.3719H186.959C186.691 47.3719 186.469 47.0889 186.469 46.7579C186.469 46.4268 186.691 46.1467 186.959 46.1467H195.781C196.053 46.1467 196.283 46.4212 196.283 46.7579Z"
                        fill="#8D93A5"
                      />
                      <path
                        d="M217.546 53.4976H165.204V143.749H217.546V53.4976Z"
                        fill="white"
                      />
                      <ellipse
                        cx="171.516"
                        cy="71.2071"
                        fill="#3F4350"
                        fill-opacity="0.24"
                        rx="3.53454"
                        ry="3.52983"
                      />
                      <ellipse
                        cx="171.516"
                        cy="92.386"
                        fill="#3F4350"
                        fill-opacity="0.24"
                        rx="3.53454"
                        ry="3.52983"
                      />
                      <ellipse
                        cx="171.516"
                        cy="113.565"
                        fill="#3F4350"
                        fill-opacity="0.24"
                        rx="3.53454"
                        ry="3.52983"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705966"
                        width="24.0349"
                        x="176.465"
                        y="69.0891"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705967"
                        width="24.0349"
                        x="176.465"
                        y="90.2681"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705967"
                        width="24.0349"
                        x="176.465"
                        y="111.447"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705966"
                        width="33.9316"
                        x="176.465"
                        y="77.5608"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705965"
                        width="33.9316"
                        x="176.465"
                        y="98.7397"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705963"
                        width="33.9316"
                        x="176.465"
                        y="119.919"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705966"
                        width="16.9658"
                        x="176.465"
                        y="81.7966"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705965"
                        width="16.9658"
                        x="176.465"
                        y="102.976"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705967"
                        width="16.9658"
                        x="176.465"
                        y="124.155"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705966"
                        width="36.7592"
                        x="176.465"
                        y="73.325"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705965"
                        width="36.7592"
                        x="176.465"
                        y="94.5039"
                      />
                      <rect
                        fill="#3F4350"
                        fill-opacity="0.24"
                        height="1.41193"
                        rx="0.705967"
                        width="36.7592"
                        x="176.465"
                        y="115.683"
                      />
                      <rect
                        fill="#1E325C"
                        height="9.80105"
                        width="52.342"
                        x="165.204"
                        y="53.4976"
                      />
                      <rect
                        fill="white"
                        height="1.22513"
                        opacity="0.4"
                        rx="0.612565"
                        width="17.1747"
                        x="173.791"
                        y="57.9897"
                      />
                      <rect
                        fill="white"
                        height="4.08377"
                        opacity="0.4"
                        rx="2"
                        width="4.08922"
                        x="210.595"
                        y="56.3562"
                      />
                      <rect
                        fill="white"
                        height="4.08377"
                        opacity="0.4"
                        rx="2"
                        width="4.08922"
                        x="168.066"
                        y="56.3562"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="clip0_754_57919"
                      >
                        <rect
                          fill="white"
                          height="142"
                          width="179"
                        />
                      </clippath>
                      <clippath
                        id="clip1_754_57919"
                      >
                        <rect
                          fill="white"
                          height="94.5349"
                          transform="translate(22.3262 10.0183)"
                          width="146.725"
                        />
                      </clippath>
                      <clippath
                        id="clip2_754_57919"
                      >
                        <rect
                          fill="white"
                          height="121"
                          transform="translate(160 38)"
                          width="63"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </div>
              </div>
              <div
                class="buttons"
              >
                <button
                  class="confirm-btn btn btn-default"
                  type="button"
                >
                  Close
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
